src/main/java/uk/gov/ons/census/fwmt/jobservice/nc/utils/NamedHouseholderRetrieval.java

```diff
@@ -0,0 +1,123 @@
+package uk.gov.ons.census.fwmt.jobservice.nc.utils;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.core.io.Resource;
+import org.springframework.stereotype.Service;
+import uk.gov.ons.census.fwmt.common.data.nc.CaseDetailsDTO;
+import uk.gov.ons.census.fwmt.common.data.nc.CaseDetailsEventDTO;
+import uk.gov.ons.census.fwmt.common.data.nc.RefusalTypeDTO;
+import uk.gov.ons.census.fwmt.common.error.GatewayException;
+import uk.gov.ons.census.fwmt.events.component.GatewayEventManager;
+
+import java.io.IOException;
+import java.time.OffsetDateTime;
+import java.util.List;
+
+import static uk.gov.ons.census.fwmt.jobservice.config.GatewayEventsConfig.UNABLE_TO_DECRYPT_NAME;
+import static uk.gov.ons.census.fwmt.jobservice.config.GatewayEventsConfig.UNABLE_TO_READ_EVENT_PAYLOAD;
+
+@Service
+public class NamedHouseholderRetrieval {
+
+  @Autowired
+  private GatewayEventManager eventManager;
+
+  @Value("${decryption.pgp}")
+  private Resource privateKey;
+
+  @Value("${decryption.password}")
+  private String privateKeyPassword;
+
+  public String getAndSortRmRefusalCases(String caseId, CaseDetailsDTO houseHolder) throws GatewayException {
```

Comment:
This method is a great candidate for using lambda expressions.Lets go through this together?